### DISCUSSION TITLE
Error in VSCode from jest-dom when using Plug'n'Play

### DISCUSSION THREAD

**Discussion by Ionshard** [yarnpkg/berry#3325] [Q&A] [answered]
I am starting a new project at work and I am interested in trying out Typescript and eventually I will want to manage multiple packages so Yarn workspaces are why I started looking into Yarn. When getting started I liked the Plug'n'Play feature and would love to use it but I am running into the following issue.

When I use the create-react-app typescript template it creates a simple test using jest-dom. However, it looks like something goes wrong with VSCode resolving correct types from jest-dom when using PnP. The tests when run via `yarn test` work fine but when I open the test in VSCode I see the following problem:

![image](https://user-images.githubusercontent.com/86419/130132759-fd887e2d-1384-4884-add0-3532df9e0ad4.png)
`Property 'toBeInTheDocument' does not exist on type 'JestMatchers<HTMLElement>'.ts(2339)`

If I switch the nodeLinker to `node-modules` that error goes away immediately.

This may not be a direct issue with yarn, it lies somewhere in the VSCode/Typescript/jest-dom/yarn-pnp landscape but I have no idea where to go from here to figure out where the issue is. Though since I am certain that yarn PnP is at least *partially* responsible I have come here.

I tried a lot of the information contained in https://github.com/testing-library/jest-dom/issues/123 but that looks like that was an issue with Typescript itself and is now resolved and you can see this is true because `yarn test` works itself.

## Steps to Reproduce
```bash
yarn create react-app demo-app --template typescript
cd demo-app
yarn set version berry
# comment out the nodeLinker line in .yarnrc.yml
yarn install
yarn dlx @yarnpkg/sdks vscode
code .
# open up src/App.test.tsx
```

**Comment by merceyz** [comment, marked as answer]
The `create-react-app` template is missing a dependency on `@types/testing-library__jest-dom`

**Reply by Ionshard** [reply]
I have added `@types/testing-library__jest-dom` as a dependency of my application and VSCode still gives me the same error. If this was just a missing dependency wouldn't I also expect to see the issue when I take VSCode out of the equation? When I run `yarn test` which will run jest everything works fine.

**Reply by merceyz** [reply]
When I ran `yarn tsc --noEmit` that is what it complained about, fixing it cleared the error from VSCode as well.

**Reply by Ionshard** [reply]
Awesome thank you that worked. My apologies I was having some issues because I had deleted the test application I had used above but forgot my actual application was in a workspace and I had the workspace open in VSCode as well. I had to add that dependency as a dependency on the workspace as well as at the package level.
